src/Ship.js

```diff
@@ -0,0 +1,28 @@
+
+function Ship(itinerary) {
+  this.itinerary = itinerary;
+  this.currentPort = itinerary.ports[0];
+  this.previousPort = null;
+  this.currentPort.addShip(this);
+};
+
+Ship.prototype.setSail = function() {
+  const itinerary = this.itinerary;
+  const currentPortIndex = itinerary.ports.indexOf(this.currentPort);
+
+  if(currentPortIndex === (itinerary.ports.length - 1)){
+      throw new Error('End of itinerary reached');
+  }
+  this.previousPort = this.currentPort;
+  this.currentPort = null;
+  this.previousPort.removeShip(this);
```

Comment:
you only use this itinerary to access the ports. You could reduce code repetition by changing like so:
```
const ports = this.itinerary.ports;
const currentPortIndex = ports.indexOf(this.currentPort);

if (currentPortIndex === (ports.length - 1)) {
...
```